### PR TITLE
readline: Update to 8.0 and various fixes

### PR DESCRIPTION
**Pull request by diizzyy**
Update (lib)readline to 8.0
Remove autoreconf
Remove blankspace at the end of the lines in description
Remove --enable-shared and --enable-static as they're enabled by default
Remove TARGET_CPPFLAGS
Simplify install sections
Install readline.pc (pkgconfig)
Add patch for linking (lib)ncurses

Source:
https://git.buildroot.net/buildroot/plain/package/readline/0000-curses-link.patch

Signed-off-by: Daniel Engberg <daniel.engberg.lists@pyret.net>

### CODE REVIEW

**Comment by diizzyy** [comment]
Compile tested on: mvebu, WRT3200ACM, OpenWrt master + sunxi, OrangePi PC, OpenWrt master
Run tested on: sunxi, OrangePi PC, OpenWrt master

File: package/libs/readline/Makefile [hunk 1]
```diff
@@ -30,45 +29,39 @@ define Package/libreadline
   SECTION:=libs
   CATEGORY:=Libraries
   TITLE:=Command lines edition library
+  DEPENDS:=+libncursesw
   URL:=http://cnswww.cns.cwru.edu/php/chet/readline/rltop.html
   ABI_VERSION:=$(PKG_VERSION)
```
Comment (jow-) [inline package/libs/readline/Makefile]: Please set the ABI version to `8`.

File: package/libs/readline/Makefile [hunk 2]
```diff
@@ -30,45 +29,39 @@ define Package/libreadline
...
-	$(CP) $(PKG_INSTALL_DIR)/usr/lib/lib{history,readline}.{a,so,so.7,so.7.0} $(1)/usr/lib/
+	$(INSTALL_DIR) $(1)/usr/lib/pkgconfig
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/lib{history,readline}.{a,so*} $(1)/usr/lib/
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/pkgconfig/readline.pc $(1)/usr/lib/pkgconfig/
 endef
 
 define Package/libreadline/install
 	$(INSTALL_DIR) $(1)/usr/lib
-	$(CP) $(PKG_INSTALL_DIR)/usr/lib/lib{history,readline}.{so,so.7,so.7.0} $(1)/usr/lib/
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/lib{history,readline}.so* $(1)/usr/lib/
```
Comment (jow-) [inline package/libs/readline/Makefile]: Please only bundle `*.so.8` and `*.so.8.0` but not the `*.so` symlink.

**Review by jow- — CHANGES_REQUESTED** [review]

File: package/libs/readline/Makefile (hunk 2 above)
Comment (diizzyy) [reply package/libs/readline/Makefile]: Done

**Comment by diizzyy** [comment]
@jow- 
Fixed

**Comment by jow-** [comment]
Pulled into my staging tree at https://git.openwrt.org/openwrt/staging/jow.git

**Pull request closed without merging**